src/constfold.c

```diff
@@ -901,6 +901,11 @@ Expression *Equal(enum TOK op, Type *type, Expression *e1, Expression *e2)
                 if (cmp == 0)
                     break;
             }
+            if (cmp && es1->type->needsNested())
+            {
+                if ((es1->sinit != NULL) != (es2->sinit != NULL))
```

Comment:
@9rnsr I'm confused by this code.  What does sinit have to do with equality comparison?

Both the tests here and the extra test in #1349 seem to still pass with this code removed.  Has this code become obsolete?  Or has something shifted so that we no longer have test coverage?